### DISCUSSION TITLE
node class list resets when selected

### DISCUSSION THREAD

**Discussion by alexpeverett** [bcakmakoglu/vue-flow#1623] [Q&A] [answered]
Hi,

I'm attempting to append a class to the class lists of nodes but whenever I do and select the node, the class list is overwritten and the appended class is removed. Is there a way I can append a class to the class list without it resetting on select?

Thanks.

**Comment by bcakmakoglu** [comment]
Without any code or reproduction I can't tell what you're doing and why it's not working.
Please provide some proper context to your question otherwise I can't help you debug the issue.

**Reply by alexpeverett** [reply]
When we have a dsl error we would like to change the colour of the node to orange (want to append class to node class list):
<img width="795" alt="Screenshot 2024-09-16 at 13 11 16" src="https://github.com/user-attachments/assets/9366b6f3-ddf4-431a-ac70-39f4766948d9">

The class appends to the class list but the issue is when a node is selected the class list is overwritten and the appended class is removed.

Unselected with class appended:
<img width="235" alt="Screenshot 2024-09-16 at 13 11 44" src="https://github.com/user-attachments/assets/2bb3b5f5-43ab-464b-aa59-3a4c3cd1ae32">
<img width="904" alt="Screenshot 2024-09-16 at 13 11 59" src="https://github.com/user-attachments/assets/5834f548-f750-4794-a94c-f30e70b15def">

Selected and classes are overwritten:
<img width="236" alt="Screenshot 2024-09-16 at 13 12 15" src="https://github.com/user-attachments/assets/1c374758-567c-4b5e-8e79-d0486b30785c">
<img width="897" alt="Screenshot 2024-09-16 at 13 12 28" src="https://github.com/user-attachments/assets/4ba0af5f-dd08-408c-b321-e8d297e1c363">

Unselected without error class:
<img width="894" alt="Screenshot 2024-09-16 at 13 12 44" src="https://github.com/user-attachments/assets/b2b8403b-7f01-412e-a6c0-771bbc50cccb">

Is there a way we can append to the class list without it getting overwritten?

**Reply by bcakmakoglu** [reply]
You have given me 0 code here, what exactly am I supposed to debug? Screenshots?

**Reply by bcakmakoglu** [reply, marked as answer]
Without knowing *how* you add/remove class names from a node, it's really hard to tell what's going on and why it's not working for you, hence why I keep asking to see some code ^^

Here's a basic example of using `node.class = 'dark'` to color a node in a dark theme, which is *not* reset when a node is selected: https://codesandbox.io/p/devbox/dazzling-bhabha-45j57j?file=%2Fsrc%2Fassets%2Findex.css%3A34%2C16

The same thing can be seen in the [Basic example](https://vueflow.dev/examples/) when using the dark mode toggle.
